[ILM] TS conversion of Index Management plugin extension

yuliacech opened the pull request:
## Summary

This PR converts Index Management plugin extension to TypeScript as an ongoing effort of converting the ILM plugin #74271. No UI/UX changes were made in this PR. To test for any regression, navigate to **Index Management** plugin and check following features in the _Indices_ tab: 

- Action button to **retry a lifecycle policy** (only shown if all indices in the list have a lifecycle phase error)
- Action button to **add a lifecycle policy** (if the index is not managed already)
- Action button to **delete a lifecycle policy** (if the index is managed)
- A banner on top of the indices list if there is an index with a lifecycle phase error (an error can be created by setting non existent snapshot policy name in Delete phase, for example)
- A section with lifecycle policy information in the Index details flyout (only shown if the index is managed)
- A filter for **lifecycle status** and **lifecycle phase** 


### Todo
After this PR and policies table conversion PR [#76006](https://github.com/elastic/kibana/pull/76006) are both merged, a follow-up PR will convert all the remaining js files, such as jest tests, services and helpers, also any remaining `any` types will be cleaned up to complete [#74271](https://github.com/elastic/kibana/issues/74271)

yuliacech committed:
[ILM] TS conversion of Index Management plugin extension

yuliacech committed:
[ILM] TS conversion of Index Management plugin extension

thompsongl committed:
Upgrade EUI to v28.2.0 (#75700)

* eui to v28.0.0

* eui to 28.2.0

* euiselectableoptions type update

* targeted nohoist

* src snapshot updates

* x-pack snapshot updates

* strong -> mark

* upgrade @elastic/charts to v21.0.1

* strong -> mark

* fix charts version merge

* maps -> add_field_tooltip_popover type update

* snapshot update

* Fix gridline visibility

Co-authored-by: Justin Kambic <justin.kambic@elastic.co>

Co-authored-by: nickofthyme <nick.ryan.partridge@gmail.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Justin Kambic <justin.kambic@elastic.co>

lcawl committed:
[ML] Remove "Are you sure" from filter lists (#76180)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

patrykkopycinski committed:
[Security Solution] Refactor FirstLastSeenHost to use Search Strategy (#76252)

chrisronline committed:
Handle cloud test failures (#74162)

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71354/)
* Commit: 8ef88417859f3a0c861c5921cd3b2af2a626ef1d

### Build metrics
#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/8ef88417859f3a0c861c5921cd3b2af2a626ef1d) | diff | [baseline](https://github.com/elastic/kibana/commit/2486a71522782d63d1691e04b96ca08a8acc3753) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 232.1KB | +1.9KB | 230.2KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71354/","number":"71354","commit":"8ef88417859f3a0c861c5921cd3b2af2a626ef1d"}]}
PIPELINE-->

mistic committed:
chore(NA): add elastic prefix to eslint-config-kibana (#76059)

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/extend_index_management/components/add_lifecycle_confirm_modal.tsx:
```diff
@@ -26,9 +28,25 @@ import {
+  selectedPolicyName: string;
+  selectedAlias: string;
+  policies: PolicyFromES[];
+  policyError?: string;
```
nit: might be more clear as `policyErrorMessage`. I know this was pre-existing, so feel free to ignore 😄

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/extend_index_management/components/index_lifecycle_summary.tsx:
```diff
@@ -176,21 +194,25 @@ export class IndexLifecycleSummary extends Component {
-        </EuiDescriptionListDescription>,
-      ];
+      const cell = (
+        <Fragment>
```
nit: you can also use `<></>` for fragments

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/extend_index_management/index.tsx:
```diff
@@ -35,22 +38,22 @@ export const retryLifecycleActionExtension = ({ indices }) => {
 };
 
-export const removeLifecyclePolicyActionExtension = ({ indices, reloadIndices }) => {
+export const removeLifecyclePolicyActionExtension = ({ indices, reloadIndices }: any) => {
```
At the very least, I think this could be improved to:

```suggestion
export const removeLifecyclePolicyActionExtension = ({ indices, reloadIndices }: {indices: any; reloadIndices: () => void}) => {
```

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/extend_index_management/index.tsx:
```diff
@@ -9,21 +9,24 @@ import { get, every, some } from 'lodash';
 import { RemoveLifecyclePolicyConfirmModal } from './components/remove_lifecycle_confirm_modal';
 
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ExtensionsSetup } from '../../../index_management/public/services';
```
I think if you export `ExtensionsSetup` from `x-pack/plugins/index_management/public/index.ts` you can remove the eslint-disable.

Commit:
[canvas] remove unused file (#76295)

Co-authored-by: spalger <spalger@users.noreply.github.com>

jloleysens committed:
[Snapshot & Restore] Decode URIs and fix editing of a policy (#76278)

* fix URI decoding and editing of a policy which backs up all indices

* fix type issue

* fix general use of encoding and update decode algo

* fix serialisation of snapshots and added a test

* Fix test description name

* Update attempt_to_uri_decode.ts

* catch errors from decoding in the already throwing code

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

patrykkopycinski committed:
[Security Solution] Refactor Network TLS to use Search Strategy (#76241)

pgayvallet committed:
replace /elasticsearch/_msearch usages with custom endpoint (#76259)

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/extend_index_management/components/index_lifecycle_summary.tsx:
```diff
@@ -80,7 +98,7 @@ export class IndexLifecycleSummary extends Component {
     this.setState({ showPhaseExecutionPopover: false });
   };
-  renderStackPopoverButton(ilm) {
+  renderStackPopoverButton(ilm: any) {
```
can we avoid using `any` here?

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/extend_index_management/components/index_lifecycle_summary.tsx:
```diff
@@ -105,10 +123,7 @@ export class IndexLifecycleSummary extends Component {
-    if (!ilm.phase_execution) {
-      return null;
-    }
+  renderPhaseExecutionPopoverButton(ilm: any) {
```
can we avoid using `any` here?

cjcenizal committed:
Fix es_ui_shared eslint violations for useRequest hook (#72947)

* Reconcile request helpers with eslint rules for React hooks.
  - Add clearer cleanup logic for unmounted components.
  - Align logic and comments in np_ready_request.ts and original request.ts.
* Reorganize modules and convert tests to TS.
  - Split request.ts into send_request.ts and use_request.ts.
  - Convert test files into TS.
  - Relax SendRequestResponse type definition to type error as any instead of expecting an Error, since we weren't actually fulfilling this expectation.
* Convert everything to hooks and add test coverage for request behavior.
* Fix Watcher memoization bugs.

alisonelizabeth reviewed (commented):
Great work @yuliacech! I tested locally and did not find any regressions.

I left a few code comments. Also, regarding the use of `any` for `indices` - could you utilize this interface in index_management: https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/server/types.ts#L29?

nreese committed:
[Maps] fix duplicate EuiPopover id's (#76202)

* [Maps] fix duplicate ID's

* tslint cleanup

* use layer id instead of layer name for popover id

* tslint fixes

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Commit:
Add cacheClear & cacheDelete methods (#76306)

YulNaumenko committed:
Added validation to display an error when creating index action in alert with invalid document. (#75929)

YulNaumenko committed:
Added deprecation for renaming config option 'xpack.actions.whitelistedHosts' to 'xpack.actions.allowedHosts' (#76325)

poffdeluxe committed:
[Canvas] Add telemetry around workpad variable usage (#75683)

* Add telemetry around workpad variable usage

* Fix typecheck

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

YulNaumenko committed:
Exposed separate from ProxySettings rejectUnauthorized configuration option. (#76061)

* Exposed separate from ProxySettings rejectUnauthorized configuration option.

* Fixed type checks

* fixed tests

pgayvallet committed:
allow multiple registration of the same feature (#76272)

pgayvallet committed:
Delete src/legacy/ui/public folder (#76085)

* delete src/legacy/ui/public folder

* remove jest.mock('ui/XXX'); from tests

* adapt stubbedLogstashIndexPatternService

* remove delete keys from translation files

* fix types import with Capabilities

* remove legacy test utils

* remove dead file referencing ui/newPlatform

* move saved-object-finder styles to timelion plugin

mdelapenya committed:
[APM UI] Rephrase BDD steps for RUM scenarios (#72492)

* chore: group tests doing the same

This will reduce the time it takes to execute the tests

* chore: rephrase step

* chore: extract common code to a function

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

pgayvallet committed:
restore dummy documentation_links.ts file to fix doc build (#76354)

angorayc committed:
[Security Solution]  Add tests for import / export timelines (#75537)

* add intergration test for install prebuilt timelines

* add integration tests

* add functional test for export timeline

* clean up

* asserts the content of the exported timeline

* update selector

* update selector

* reuses the timeline id from the expected exported file

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Gloria Hornero <snootchie.boochies@gmail.com>

yuliacech commented:
@elasticmachine merge upstream

pgayvallet committed:
xpack_main legacy plugin pre-removal cleanup (#76257)

* cleanup xpack_main legacy plugin, remove capabilities mixin

* fix test env

* delete injectXpackSignature and related tests

yuliacech committed:
Merge remote-tracking branch 'origin/ilm_index_management' into ilm_index_management

# Conflicts:
#	x-pack/plugins/snapshot_restore/public/application/lib/index.ts

yuliacech commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into ilm_index_management

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71518/)
* Commit: e6e59392bb4f3757b82f5c4e9b2b951fdf8ae029
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71518/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71518/execution/node/124/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71518/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#76308](https://github.com/elastic/kibana/pull/76308)@[e6e5939](https://github.com/elastic/kibana/commit/e6e59392bb4f3757b82f5c4e9b2b951fdf8ae029)]

### History
* :green_heart: [Build #71354](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71354/) succeeded 8ef88417859f3a0c861c5921cd3b2af2a626ef1d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71354/","number":"71354","commit":"8ef88417859f3a0c861c5921cd3b2af2a626ef1d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71518/","number":"71518","commit":"e6e59392bb4f3757b82f5c4e9b2b951fdf8ae029"}]}
PIPELINE-->
